File: javascript/space-age/space-age.js
var SpaceAge = function(seconds) {
    this.seconds = seconds
};

SpaceAge.prototype.onEarth = function() {
    return Math.round(this.seconds / 60 / 60 / 24 / 365.25 * 100) / 100
};

SpaceAge.prototype.onMercury = function() {
    return Math.round(this.onEarth() / 0.2408467 * 100) / 100 // Mercury solar year 0.2408467/1 of earth's
}

SpaceAge.prototype.onVenus = function() {
    return Math.floor(this.onEarth() / 0.61519726 * 100) / 100 // Venus solar year 0.61519726/1 of earth's
}

SpaceAge.prototype.onMars = function() {
    return Math.round(this.onEarth() / 1.8808158 * 100) / 100 // Mars solar year 1.8808158/1 of earth's
}

SpaceAge.prototype.onJupiter = function() {
    return Math.round(this.onEarth() / 11.862615 * 100) / 100 // Jupiter solar year 11.862615/1 of earth's
}

SpaceAge.prototype.onSaturn = function() {
    return Math.round(this.onEarth() / 29.447498 * 100) / 100 // Saturn solar year 29.447498/1 of earth's
}

SpaceAge.prototype.onUranus = function() {
    return Math.round(this.onEarth() / 84.016846 * 100) / 100 // Uranus solar year 84.016846/1 of earth's
}

SpaceAge.prototype.onNeptune = function() {
    return Math.round(this.onEarth() / 164.79132 * 100) / 100 // Neptune solar year 164.79132/1 of earth's
}

module.exports = SpaceAge;

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {
    private Double mSeconds;
    private static final Long EARTH_YEAR_IN_SECONDS = 31557600L;
    private static final Double MERCURY_FRACTION_OF_EARTH = 0.2408467;
    private static final Double VENUS_FRACTION_OF_EARTH = 0.61519726;
    private static final Double MARS_FRACTION_OF_EARTH = 1.8808158;
    private static final Double JUPITER_FRACTION_OF_EARTH = 11.862615;
    private static final Double SATURN_FRACTION_OF_EARTH = 29.447498;
    private static final Double URANUS_FRACTION_OF_EARTH = 84.016846;
    private static final Double NEPTUNE_FRACTION_OF_EARTH = 164.79132;
    

    SpaceAge(double seconds) {
        this.mSeconds = seconds;
    }

    double getSeconds() {
        return mSeconds;
    }

    double onEarth() {
        return mSeconds / EARTH_YEAR_IN_SECONDS;
    }

    double onMercury() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * MERCURY_FRACTION_OF_EARTH);
    }

    double onVenus() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * VENUS_FRACTION_OF_EARTH);
    }

    double onMars() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * MARS_FRACTION_OF_EARTH);
    }

    double onJupiter() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * JUPITER_FRACTION_OF_EARTH);
    }

    double onSaturn() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * SATURN_FRACTION_OF_EARTH);
    }

    double onUranus() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * URANUS_FRACTION_OF_EARTH);        
    }

    double onNeptune() {
        return mSeconds / (EARTH_YEAR_IN_SECONDS * NEPTUNE_FRACTION_OF_EARTH);        
        
    }

}
